src/en/koharu/src/eu/kanade/tachiyomi/extension/en/koharu/Koharu.kt

```diff
@@ -0,0 +1,303 @@
+        val images = imagesResponse.parseAs<ImagesInfo>()
+        return images
+    }
+
+    // Latest
+
+    override fun latestUpdatesRequest(page: Int) = GET("$apiBooksUrl?page=$page", headers)
+    override fun latestUpdatesParse(response: Response) = popularMangaParse(response)
+
+    // Popular
+
+    override fun popularMangaRequest(page: Int) = GET("$apiBooksUrl?sort=6&page=$page", headers)
+    override fun popularMangaParse(response: Response): MangasPage {
+        val data = response.parseAs<Books>()
+
+        return MangasPage(data.entries.map(::getManga), data.page * data.limit < data.total)
+    }
+
+    // Search
+
+    override fun getFilterList(): FilterList = getFilters()
+
+    override fun fetchSearchManga(page: Int, query: String, filters: FilterList): Observable<MangasPage> {
+        return when {
+            query.startsWith(PREFIX_ID_KEY_SEARCH) -> {
+                val ipk = query.removePrefix(PREFIX_ID_KEY_SEARCH)
+                val response = client.newCall(GET("$apiBooksUrl/detail/$ipk", headers)).execute()
+                Observable.just(searchMangaParse2(response))
+            }
+            else -> super.fetchSearchManga(page, query, filters)
+        }
+    }
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        val url = apiBooksUrl.toHttpUrl().newBuilder().apply {
+            val terms = mutableListOf(query.trim())
+
+            filters.forEach { filter ->
+                when (filter) {
+                    is SortFilter -> addQueryParameter("sort", filter.getValue())
+
+                    is CategoryFilter -> {
+                        val activeFilter = filter.state.filter { it.state }
+                        if (activeFilter.isNotEmpty()) {
+                            addQueryParameter("cat", activeFilter.sumOf { it.value }.toString())
+                        }
+                    }
+
+                    is TextFilter -> {
+                        if (filter.state.isNotEmpty()) {
+                            terms += filter.state.split(",").filter(String::isNotBlank).map { tag ->
+                                val trimmed = tag.trim()
+                                buildString {
+                                    if (trimmed.startsWith('-')) {
+                                        append("-")
+                                    }
+                                    append(filter.type)
+                                    append("!:")
+                                    append("\"")
+                                    append(trimmed.lowercase().removePrefix("-"))
+                                    append("\"")
+                                }
+                            }
+                        }
+                    }
+                    else -> {}
+                }
+            }
+            addQueryParameter("s", "title:\"$query \" " + terms.joinToString(" "))

```

Comment:
If terms is empty it will have a trailing space. Might be used for fingerprinting.